base_layer/p2p/src/initialization.rs

```diff
@@ -108,6 +108,23 @@ pub struct CommsConfig {
     pub listener_liveness_allowlist_cidrs: Vec<String>,
 }
 
+impl Default for CommsConfig {
+    fn default() -> Self {
+        Self {
+            datastore_path: Default::default(),
```

Comment:
I think the Default derive macro does exactly the same as this implementation

```rust
#[derive(Clone, Default)]
pub struct CommsConfig {
```

Reply:
Because there is a NodeIdentity in this config struct, you cannot (should not) use a Default impl - comms config probably shouldnt have node identity in it as it's not really a config struct (it should be passed separately IMO). That refactor is out of scope for this PR though.

Reply:
I agree with @sdbondi, not a lot of these fields make sense as a default, so rather take the Default impl out. Is there a particular reason you needed the Default?

Reply:
Thanks, good comment, makes sense. Fixing it now.